common/src/auth/interceptors/token-expired.ts

```diff
@@ -1,107 +1,45 @@
 import { AbstractUser } from '../../user';
-import { ApiService } from '../../api';
 import { AuthService } from '../auth.service';
 import { JwtExceptions } from '../enums';
-import {
-  catchError,
-  filter,
-  mapTo,
-  switchMap,
-  take,
-  withLatestFrom
-} from 'rxjs/operators';
-import { EMPTY, Observable, of, throwError } from 'rxjs';
+import { catchError, switchMap } from 'rxjs/operators';
+import { Observable, throwError } from 'rxjs';
 import {
   HttpErrorResponse,
   HttpEvent,
   HttpHandler,
   HttpInterceptor,
-  HttpRequest
+  HttpRequest,
+  HttpStatusCode
 } from '@angular/common/http';
 import { Injectable } from '@angular/core';
-import { isNull } from 'lodash';
-import { JwtHelperService, JwtInterceptor } from '@auth0/angular-jwt';
 import { AuthConfig } from '../config';
 
 @Injectable()
 export class TokenExpiredInterceptor implements HttpInterceptor {
   constructor(
-    private apiService: ApiService,
     private authConfig: AuthConfig,
-    private authService: AuthService<AbstractUser>,
-    private jwtHelperService: JwtHelperService,
-    private jwtInterceptor: JwtInterceptor
+    private authService: AuthService<AbstractUser>
   ) { }
 
   public intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
-    if (!this.shouldInterceptRequest(request)) {
-      return next.handle(request);
-    }
-
-    return this.authService.token$
+    return next
+      .handle(request)
       .pipe(
-        take(1),
-        switchMap((token) => {
-          if (!this.shouldRefreshToken(token) || this.isRefreshTokenRequest(request)) {
-            return next.handle(request);
-          }
-
-          return this.refreshTokenAndHandleRequest(request, next);
-        }),
         catchError((response: HttpErrorResponse) => {
-          if (response.status === 401) {
-            if (response.error?.error === JwtExceptions.TOKEN_EXPIRED) {
-              return this.refreshTokenAndHandleRequest(request, next);
-            }
-
-            this.authService.unauthorize();
+          if (
+            response.status === HttpStatusCode.Unauthorized &&
+            response.error.error === JwtExceptions.TOKEN_EXPIRED &&
+            !this.authConfig.unauthorizedRoutes.includes(response.url)
+          ) {
+            return this.authService
+              .refreshToken()
+              .pipe(
+                switchMap(() => next.handle(request))

```

Comment:
Not sure, but shouldn't we also call `this.authService.unauthorize();` in case of token refresh failure as this was in the former implementation?